modules/@apostrophecms/asset/index.js

```diff
@@ -28,72 +28,111 @@ module.exports = {
   },
 
   init(self, options) {
-
     self.restartId = self.apos.util.generateId();
-
-    self.addTask('build', 'Build Apostrophe frontend javascript master import files', async function (apos, argv) {
```

Comment:
I refactored the body of the task to a method so it can be properly overridden. No code changes beyond that except those specific to icons.